rest_framework/authtoken/views.py

```diff
@@ -2,18 +2,20 @@
 from rest_framework.authtoken.models import Token
 from rest_framework.authtoken.serializers import AuthTokenSerializer
 from rest_framework.compat import coreapi, coreschema
+from rest_framework.generics import GenericAPIView
 from rest_framework.response import Response
 from rest_framework.schemas import ManualSchema
-from rest_framework.views import APIView
+from rest_framework.schemas import coreapi as coreapi_schema
 
 
-class ObtainAuthToken(APIView):
+class ObtainAuthToken(GenericAPIView):
```

Comment:
Instead of this, can you just define `get_serializer()`, and then use that in `post()`?

Reply:
Definitely! This is done :)